test/unit/mdc-textfield/mdc-text-field.test.js

```diff
@@ -96,17 +133,50 @@ test('#destroy cleans up the ripple if present', () => {
   td.verify(component.ripple.destroy());
 });
 
+test('#destroy cleans up the bottom line if present', () => {
+  const root = getFixture();
+  const component = new MDCTextField(root, undefined, undefined, (el) => new FakeBottomLine(el));
+  component.destroy();
+  td.verify(component.bottomLine_.destroy());
+});
+
+test('#destroy cleans up the helper text if present', () => {
+  const root = getFixture();
+  root.querySelector('.mdc-text-field__input').setAttribute('aria-controls', 'helper-text');
+  const helperText = getHelperTextElement();
+  document.body.appendChild(helperText);
+  const component =
+    new MDCTextField(root, undefined, undefined, undefined, (el) => new FakeHelperText(el));
```

Comment:
Should we bake these other fake factories into `setupTest` so that we can then reference them directly from its return map rather than needing to dig into arguably internal instance properties (e.g. `helperText_` here, `icon_` and `label_` below)?

Reply:
Good call, done!